core/modules/user/user.theme.inc

```diff
@@ -386,3 +386,63 @@ function template_preprocess_user_profile(&$variables) {
   $variables['theme_hook_suggestions'][] = 'user_profile__' . $account->uid;
   $variables['theme_hook_suggestions'][] = 'user_profile__' . $account->uid . '__' . $view_mode;
 }
+
+/**
+ * Process variables for user-simplified-page.tpl.php.
+ *
+ * The $variables array contains the following arguments:
+ * - $element
+ *
+ * @see user-simplified-page.tpl.php
+ */
+function template_preprocess_user_simplified_page(&$variables) {
```

Comment:
Isn't that supposed to be `user_preprocess_user_simplified_page`? It implements `hook_preprocess_HOOK()`.

I know, it's used that way (literal `template_preprocess_...` all over the place in core, so maybe I'm wrong. I'm assuming, both works.

Reply:
Both works. Because of the order of invocations, we want to user `template_` for this one, since we always need this to be invoked first. Typically, AFAIK, the "first and essential" preprocess function call starts with `template_preprocess...`. Other modules that want to add variables or make changes can then user `MYMODULE_preprocess...` . But maybe I'm wrong?

Reply:
You're absolutely right, many thanks for explaining - and I learned something new again. :grinning: 

Found it in the documentation for [function theme](https://docs.backdropcms.org/api/backdrop/core%21includes%21theme.inc/function/theme/1).
Consider this comment thread as resolved. :+1:

Reply:
+1 @argiepiano is correct in the naming here.